ts_src/psbt.ts

```diff
@@ -1746,49 +1736,110 @@ function randomBytes(options?: RngOpts): Buffer {
   return rng(32);
 }
 
-interface BlindingData {
-  value: string;
-  ag: Buffer;
-  abf: Buffer;
-  vbf: Buffer;
+export interface BlindingData {
+  satoshis: number;
+  amountBlinder: string;
+  asset: string;
+  assetBlinder: string;
 }
 
-function getBlindingDataForInput(
-  prevout: WitnessUtxo,
-  blindPrivKey?: Buffer,
+// Buffer = privateBlindingKey for conf inputs
+// BlindingData = blinders for already unblinded conf inputs
+// undefined = unconfidential inputs
+export type BlindingDataLike = Buffer | BlindingData | undefined;
+
+/**
+ * Compute outputs blinders
+ * @param inputsBlindingData the transaction inputs blinding data
+ * @param outputsData data = [satoshis, asset] of output to blind
+ * @returns an array of BlindingData[] corresponding of blinders to blind outputs specified in outputsData
+ */
+export function computeOutputsBlindingData(
+  inputsBlindingData: BlindingData[],
+  outputsData: Array<[number, string]>,
+): BlindingData[] {
+  const outputsBlindingData: BlindingData[] = [];
+  outputsData.forEach(([satoshis, asset], outputIndex) => {
+    const blindingData: BlindingData = {
+      satoshis,
+      asset,
+      assetBlinder: randomBlinder(),
+      amountBlinder: '', // invalid blinder at this step
+    };
+
+    if (outputIndex === outputsData.length - 1) {
+      // tslint:disable-next-line:no-shadowed-variable
+      const inputsValues = inputsBlindingData.map(({ satoshis }) =>
+        satoshis.toString(10),
+      );
+      const outputsValues = outputsData
+        .map(([amount]) => amount.toString(10))
+        .concat(satoshis.toString(10));
+      const inputsAssetBlinders = inputsBlindingData.map(({ assetBlinder }) =>
+        Buffer.from(assetBlinder, 'hex'),
+      );
+      const outputsAssetBlinders = outputsBlindingData
+        .map(({ assetBlinder }) => Buffer.from(assetBlinder, 'hex'))
+        .concat(Buffer.from(blindingData.assetBlinder, 'hex'));
+      const inputsAmountBlinders = inputsBlindingData.map(({ amountBlinder }) =>
+        Buffer.from(amountBlinder, 'hex'),
+      );
+      const outputsAmountBlinders = outputsBlindingData.map(
+        ({ amountBlinder }) => Buffer.from(amountBlinder, 'hex'),
+      );
+
+      const finalAmountBlinder = confidential
+        .valueBlindingFactor(
+          inputsValues,
+          outputsValues,
+          inputsAssetBlinders,
+          outputsAssetBlinders,
+          inputsAmountBlinders,
+          outputsAmountBlinders,
+        )
+        .toString('hex');
+
+      blindingData.amountBlinder = finalAmountBlinder;
+    } else {
+      blindingData.amountBlinder = randomBlinder();
+    }
+
+    outputsBlindingData.push(blindingData);
+  });
+
+  return outputsBlindingData;
+}
+
+export function toBlindingData(
```

Comment:
same for this, it's used only internally by rawBlindOutputs. Can we avoid to export this?